tests/baselines/reference/moduleResolution/non-relative-uses-originalPath-for-caching.js

```diff
@@ -31,8 +31,7 @@ Resolution:: {
     "/sub/dir/node_modules/@types/a/index.d.ts",
     "/sub/node_modules/a.ts",
     "/sub/node_modules/a.tsx",
-    "/sub/node_modules/a.d.ts",
-    "/sub/node_modules/a/index"
```

Comment:
We still want these paths in the failed lookup locations, yeah? They're places we need to watch for changes that could affect resolution, right? And since if a directory appears at this path, it'd potentially load something, we do actually need to monitor it?

Reply:
No—`/sub/node_modules/a.d.ts` is still present, but we removed `/sub/node_modules/a/index` because we wouldn’t have loaded it even if it existed, since it has no extension. `/sub/node_modules/a/index` as a directory would also not be relevant here, since the original request was for `"a"`.

Reply:
Right, it was for `"a"`, which found the package json at `/sub/node_modules/a/` and opened it up and found a `main`, which in turn is looking for `./index` (which is ultimately satisfied by `/sub/node_modules/a/index.ts`), _which could also be satisfied by a directory `/sub/node_modules/a/index` with a package.json in it_. That's the major difference between `main` resolution and exports - exports ignores package files in directories you point at because it's supposed to only look at specific files, while `main` can end up recursively looking up multiple `package.json`s that all refer to one another.

Reply:
The `/sub/node_modules/a/index` you see removed here was tried as a _file_ (again, for no reason—we would never load it), and still would have failed on that lookup if it existed as a directory. `/sub/node_modules/a/index.ts` _did_ exist and has higher precedence than anything inside the hypothetical `/sub/node_modules/a/index/` directory.

Only if `index.ts`, `index.tsx`, and `index.d.ts` all were missing would we then proceed into `/sub/node_modules/a/index/`, if it existed. In there, we would look for a package.json, or more index files, and you would see _those_ file names (e.g. `/sub/node_modules/a/index/package.json` appear here in `failedLookupLocations`. `failedLookupLocations` does not contain directory paths, but even if it did, it should not appear in this particular test case because we successfully resolved something at a higher priority than anything in such a directory would be.

Reply:
Oh, you're right, these are all file locations. I guess even though node would load an extensionless file at that location, we wouldn't since we don't know if it's js or not. But it feels like that aughta block resolution, regardless, rather than fallthrough.

Thinking about it, we technically support declaration files for file with arbitrary extensions - would an extensionless `index` file just a `index.d..ts` file, then?

Reply:
That might be something to explore. And yeah, I’ve brought up something similar in the past—there are many cases where we continue module resolution after Node.js would have stopped, often because (even before this PR, in other lookup paths) we don’t check for file existence of stuff that we know we can’t load. For example, if the request is for `"./foo.css"`, in CJS mode, we’ll look for `./foo.d.css.ts`, but if it doesn’t exist, we move onto trying `./foo.css.{ts,tsx,d.ts,js}`, even though we really shouldn’t if `/foo.css` itself exists.